app/controllers/api/assignments_controller.rb

```diff
@@ -0,0 +1,133 @@
+class Api::AssignmentsController < ApplicationController
+    before_action :set_assignment, only: [:show, :update, :destroy]
+    before_action :set_template, only: [:show_template, :update_template, :destroy_template]
+    respond_to :json
+
+    def create
+        @action_item = authorize ActionItem.new(action_item_params)
+        @action_item[:is_template] = false
+
+        if @action_item.save
+            created_assignments = []
+            prepare_bulk_assignment(assigned_to_ids, @action_item.id).each do |assignment_params|
+                @assignment = authorize Assignment.new(assignment_params)
+                if @assignment.save
+                    created_assignments.append(@assignment)
+                else
+                    @action_item.destroy
+                    render json: { error: 'Could not create action item' }, status: :unprocessable_entity
+                    return 
+                end
+            end
+            render json: created_assignments, status: :created
+        else
+            render json: { error: 'Could not create action item' }, status: :unprocessable_entity
+        end
+    end 
+
+    def show
+        authorize @assignment 
+        render json: @assignment, status: :ok
+    end
+
+    def update
+        authorize @assignment
+        if @assignment.update(assignment_params) && @assignment.action_item.update(action_item_params)
+            render json: @assignment, status: :ok
+        else
+            render json: { error: 'Could not update action item' }, status: :unprocessable_entity
+        end
+    end
+
+    def destroy
+        authorize @assignment
+        action_item = @assignment.action_item
+        if @assignment.destroy
+            if action_item.assignments.empty?
```

Comment:
do we still need to check this or can we just destroy the action item once assignment.destroy is successful

Reply:
we need to check this